exercises/danny_hindson/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,42 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "Circle.h"
+#include "Rectangle.h"
+#include "ShapeSorter.h"
+#include "Shape.h"
+#include "Square.h"
+#include "Triangle.h"
+#include <iostream>
+#include <string>
+#include <vector>
+
+int main(int, char **)
+{
+  Square sq(2.5);
+  Rectangle r(3, 7);
+  Circle c(4.1);
+  Triangle t(1, 10);
+
+  // create vector of pointers to Shape rather than actual Shapes so the polymorphism works
+  std::vector<Shape*> shapeVector;
+
+  shapeVector.push_back(&sq);
```

Comment:
Good! People seem to often forget that you can take the address of a stack variable and you don't have to use `new`!